src/Bicep.LangServer/Completions/BicepCompletionProvider.cs

```diff
@@ -108,6 +109,32 @@ private IEnumerable<CompletionItem> GetParamAssignmentCompletions(ParamsSemantic
             bool IsParamAssigned(ParameterSymbol declaration) => paramsSemanticModel.ParamBinder.ParamFileSymbol.ParameterAssignmentSymbols.Any(paramDeclaration => paramDeclaration.Name == declaration.Name);
         }
 
+        private IEnumerable<CompletionItem> GetParamAllowedValueCompletions(ParamsSemanticModel paramsSemanticModel, ParamsCompletionContext paramsCompletionContext)
+        {
+
+            if(paramsCompletionContext.Kind.HasFlag(ParamsCompletionContextKind.ParamValue) && paramsSemanticModel.BicepCompilation is {} bicepCompilation)
+            {
+                var bicepSemanticModel = bicepCompilation.GetEntrypointSemanticModel();
+                var bicepFileParamDeclaration = bicepSemanticModel.Parameters.Where(parameterMetaData => parameterMetaData.Name == paramsCompletionContext.ParamAssignment).FirstOrDefault();
+                if(bicepFileParamDeclaration?.TypeReference.Type is UnionType union)
+                {
+                    foreach(var typeRef in union.Members)
+                    {
+                        if(typeRef.Type is StringLiteralType stringLiteral)
+                        {
```

Comment:
There may be other type-based completions we want to add (which would necessitate generalizing this sequence), but that doesn't need to be done in this PR.